router/customdestinationmanager/customdestinationmanager.go

```diff
@@ -121,12 +122,13 @@ func (customManager *CustomManagerT) newClient(destID string) error {
 	return err
 }
 
-func (customManager *CustomManagerT) send(jsonData json.RawMessage, destType string, client, config interface{}) (int, string) {
+func (customManager *CustomManagerT) send(jsonData json.RawMessage, destType string, client interface{}, config map[string]interface{}) (int, string) {
 	var statusCode int
 	var respBody string
 	switch customManager.managerType {
 	case STREAM:
-		statusCode, _, respBody = streammanager.Produce(jsonData, destType, client, config)
+		streamProducer, _ := client.(common.StreamProducer)
```

Comment:
why did we have to move this type assertion out of `streammanager.Produce` and why we no longer test that the type assertion is a valid one?

Reply:
You mean, we should check for err rather than ignoring it? in the stream manager want to avoid taking interface{} but  we should test here too.

Reply:
Currently decided to not handle this as this is not going to happen in product at all as SendData won't even call send if client is not initialised.